File: backend/ItForum/Data/Domains/Post.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using ItForum.Data.Entities;

namespace ItForum.Data.Domains
{
    public class Post : PostEntity
    {
        [ForeignKey(nameof(ThreadId))]
        public Thread Thread { get; set; }

        [ForeignKey(nameof(CreatedById))]
        public User CreatedBy { get; set; }

        [ForeignKey(nameof(ApprovalStatusModifiedById))]
        public User ApprovalStatusModifiedBy { get; set; }

        [ForeignKey(nameof(ParentId))]
        public List<Post> Replies { get; set; }
    }
}

File: frontend/src/app/models/post.ts
import { User } from './user';
import { Timestamp } from './timestamp';

export class Post extends Timestamp {
  id: number;
  point = 0;
  content: string;
  threadId: number;
  userId: number;
  createdBy: User;
  parentId: number;
  replies: Post[];

  constructor(obj?) {
    super(obj);
    if (obj) {
      this.id = obj.id;
      this.content = obj.content;
      this.threadId = obj.threadId;
      this.userId = obj.userId;
      this.parentId = obj.parentId;
      this.replies = obj.replies;
      this.createdBy = obj.createdBy;
    }
  }
}
